### ISSUE TITLE
custom markdown renderer

### ISSUE THREAD

**Issue by thomasheyenbrock** [graphql/graphiql#2584]
Rendering Markdown content (for description texts) is currently baked into `@graphiql/react`. There we have an instance of `markdown-it` that takes care of all the markdown rendering.

@acao mentioned [here](https://github.com/graphql/graphiql/pull/2582#discussion_r928282116) that supporting custom markdown renders might be a valuable addition. If you stumble upon this issue, feedback is very welcome as we haven't yet seen any issue around this in our repo!

In terms of code structure, we could also move the `markdown-it` instance to `@graphiql/toolkit` as rendering markdown is not React specific.

**Comment by acao** [comment]
indeed, or at least supporting the ability to supply a `markdown-it` config with your own plugins. our config is nice and simple, but others may just want to provide their own `markdown-it` plugins.

that said, that leaves out the `remark` folks and others, so maybe a custom renderer option is in order!

**Comment by acao** [comment]
`monaco-graphql` note for 3.0.0: monaco does it's own markdown rendering for hover, autocomplete description, definition lookup, etc. We supply the raw markdown descriptions from the schema in different ways between graphql and variables json files, which use json schema
